### PR TITLE
Make swift-format work

### PR DESCRIPTION
**Pull request by lucteo**
Link to a swift-format that has controlled dependencies, forked inside our val-lang group.

Add CI job to run swift-format.

Change the documentation on using swift-format.

### CODE REVIEW

File: .github/workflows/format-check.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      with:
+        swift-version: "5.7"
+
+    - uses: actions/checkout@v3
+
+    - name: Swift version
+      run: swift --version
+
+    - name: Check code format
+      run: swift package plugin lint-source-code --configuration .swift-format.json --target CLI --target ValTests --target UtilsTests
```
Comment (dabrahams) [inline .github/workflows/format-check.yml:37]: Looks like you're missing many of the targets declared in Package.swift?

File: Docs/CodeFormatting.md [hunk 2]
```diff
@@ -40,12 +35,4 @@ The default keyboard shortcut for running the formatter for the open document is
...
-XCode doesn't provide a good way to install custom code formatters.
-Here is a workaround to fix formatting while building the project:
-* Go to `Product` → `Scheme` → `Edit Scheme...` in the XCode menu
-* Under `Build`, select `Post-build actions`
-* Click on `+` button, then select `New Run Script Action`
-* In the box corresponding to the script, paste the following:
-```bash
-${WORKSPACE_PATH}/../../../Tools/run-swift-format.sh fix
-```
+The `lint-source-code` and `format-source-code` can be directly applied from XCode. Right-click on the `val` package in the "Project navigator" tab, and select "Lint Source Code" or "Format Source Code". Ensure to add `--configuration .swift-format.json` inside "Arguments" expansion area, select the targets to run on, and then hit "Run".
```
Comment (dabrahams) [inline Docs/CodeFormatting.md:38]:
```suggestion
The `lint-source-code` and `format-source-code` commands can be directly applied from XCode. Right-click on the `val` package in the "Project navigator" tab, and select "Lint Source Code" or "Format Source Code". Ensure to add `--configuration .swift-format.json` inside "Arguments" expansion area, select the targets to run on, and then hit "Run".
```

Please insert line breaks in long paragraphs for diff-friendliness.

**Review by dabrahams — APPROVED** [review]
A couple of comments inline

File: Docs/CodeFormatting.md [hunk 3]
```diff
@@ -5,30 +5,25 @@ The Swift source code in this repository is meant to be checked with `swift-form
...
-At this point we use the version `swift-format` at commit SHA 0bc2f0381c72d66a949254af22208a81377cf717.
+We use the version `swift-format` that is forked in our group: https://github.com/val-lang/swift-format.
+
 
 ## Manually Running `swift-format`
 
 To check the style for the Swift code in the repository, run from root directory:
 ```bash
-Tools/run-swift-format.sh lint
+swift package plugin lint-source-code --configuration .swift-format.json --target CLI --target ValTests --target UtilsTests
```
Comment (nickpdemarco) [inline Docs/CodeFormatting.md:15]: Especially if we're going to add targets to this line before submitting, I think this should remain as a utility script. Typing out that whole invocation to format my code will be a big nuisance.

Comment (dabrahams) [reply Docs/CodeFormatting.md:15]: +1. And I want the script with no options to do a reformat followed by a lint

File: .github/workflows/format-check.yml (hunk 1 above)
Comment (lucteo) [reply .github/workflows/format-check.yml:37]: Hmmm. There are only 3 targets that appear in XCode when I run this. I don't know why it's that, but I will correct this and apply the suggestion to put the targets into the script

**Comment by lucteo** [comment]
Note: the performance of this script seems to be terrible on my machine. It takes about 45s for a fix_lint.
I'll give a short try to see if I can improve on this. But, I'll gladly take the 45s over having no CI job.

**Comment by lucteo** [comment]
Performance issues were fixed

File: .github/workflows/format-check.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
+name: Format check
+
+on:
+  push:
+    branches: [ main ]
+    paths-ignore:
+      - "Docs/**"
+      - "**.md"
+      - "README.md"
+      - "LICENSE"
```
Comment (kyouko-taiga) [inline .github/workflows/format-check.yml:10]: LICENSE must be explicitly ignored but not CODE_OF_CONDUCT or CONTRIBUTING?

**Review by kyouko-taiga — APPROVED** [review]
LGTM

**Review by nickpdemarco — APPROVED** [review]

**Pull request merged**